pkg/server/controllers.go

```diff
@@ -1478,3 +1488,188 @@ func (s *Server) WaitForSync(stop <-chan struct{}) error {
 		return nil
 	}
 }
+
+func (s *Server) addIndexersToInformers(ctx context.Context) map[schema.GroupVersionResource]replication.ReplicatedGVR {
+	// workspace indexers
+	indexers.AddIfNotPresentOrDie(s.KcpSharedInformerFactory.Tenancy().V1alpha1().Workspaces().Informer().GetIndexer(), cache.Indexers{
+		workspace.Unschedulable: workspace.IndexUnschedulable,
+	})
+	indexers.AddIfNotPresentOrDie(s.CacheKcpSharedInformerFactory.Core().V1alpha1().Shards().Informer().GetIndexer(), cache.Indexers{
+		workspace.ByBase36Sha224Name: workspace.IndexByBase36Sha224Name,
+	})
+	indexers.AddIfNotPresentOrDie(s.CacheKcpSharedInformerFactory.Tenancy().V1alpha1().WorkspaceTypes().Informer().GetIndexer(), cache.Indexers{
+		indexers.ByLogicalClusterPathAndName: indexers.IndexByLogicalClusterPathAndName,
+	})
+	// APIBinding indexers
+	indexers.AddIfNotPresentOrDie(s.KcpSharedInformerFactory.Apis().V1alpha1().APIBindings().Informer().GetIndexer(), cache.Indexers{
+		indexers.APIBindingsByAPIExport: indexers.IndexAPIBindingByAPIExport,
+	})
```

Comment:
I'm concerned about this being very hard to trace back to the respective controller that need this. Is there any chance we can re-organise the code into the respective controllers (where possible) and call that from here?

Reply:
@embik probably we can have this in each controllers , but have a condition in installControllers , like bootstrapIndexers if true , we would activate this codepath , if not we would skip it , as this is already bootstrapped? , or probably thinking of getting indexers before calling indexersAddition

Reply:
If each controller package had a `InstallIndexers` func to call in `addIndexersToInformers`, I think that would already be sufficient.

Reply:
Addressed the comment as mentioned above @embik , PTAL